### ISSUE TITLE
Update mdb_v8 for V8 4.6.x

### ISSUE THREAD

**Issue by misterdjules** [joyent/mdb_v8#36]
nodejs/node [will likely ship node v5.x with V8 4.6.x](https://github.com/nodejs/node/issues/3000), and currently mdb_v8's tests (from a build of https://github.com/joyent/mdb_v8/pull/33, that is with mdb_v8 updates for V8 4.5.x) do not pass with that version of V8:
```
[root@dev ~/mdb_v8]# node -p 'process.versions'
{ http_parser: '2.5.0',
  node: '5.0.0-pre',
  v8: '4.6.85.19',
  uv: '1.7.4',
  zlib: '1.2.8',
  ares: '1.10.1-DEV',
  modules: '46',
  openssl: '1.0.2d' }
[root@dev ~/mdb_v8]# make test
tools/catest -a
Configuration:
    SRC:                         /root/mdb_v8
    Output directory:            /var/tmp/catest.89202
    Temp directory:              /var/tmp/catest.89202_tmpfiles
    Keep successful test output: false
    Found 4 test(s) to run

===================================================

Executing test test/standalone/tst.jsclosure.js ... success.
Executing test test/standalone/tst.postmortem_details.js ... FAILED.
>>> failure details in /var/tmp/catest.89202/failure.0

Executing test test/standalone/tst.postmortem_findjsobjects.js ... FAILED.
>>> failure details in /var/tmp/catest.89202/failure.1

Executing test test/standalone/tst.postmortem_jsstack.js ... success.

===================================================

Results:
	Tests passed:	 2/ 4
	Tests failed:	 2/ 4

===================================================
Cleaning up output from successful tests ... done.
GNUmakefile:120: recipe for target 'test' failed
make: *** [test] Error 2
[root@dev ~/mdb_v8]#
```

It seems that these failures are due to [recent changes in the objects' Map data structure to replace the `inobject_properties` field with `inobject_properties_or_constructor_function_index`](https://codereview.chromium.org/1276533003):
```
[root@dev ~/mdb_v8]#`  cat /var/tmp/catest.89323/failure.0/89323.out 
mdb stderr: mdb: couldn't find class "Map", field "inobject_properties"
mdb: failed to autoconfigure from target; commands may have incorrect results!

mdb stderr: mdb: 3af1f096fd1: didn't find expected class

mdb stderr: mdb: 3af1f096fd1: didn't find expected class

mdb stderr: mdb: 3af1f096fd1: didn't find expected class

[root@dev ~/mdb_v8]# cat /var/tmp/catest.89323/failure.1/89323.out 
mdb stderr: mdb: couldn't find class "Map", field "inobject_properties"
mdb: failed to autoconfigure from target; commands may have incorrect results!

[root@dev ~/mdb_v8]#
```

There could be other breaking changes in V4.6.x, but it's a starting point.

**Comment by misterdjules** [comment]
Metadata was updated with https://codereview.chromium.org/1308113007.

**Comment by davepacheco** [comment]
Merged via #37.
